storybook/qmlTests/tests/tst_SwapModal.qml

```diff
@@ -1962,5 +1962,104 @@ Item {
 
             closeAndVerfyModal()
         }
+
+        function test_exchange_rate() {
+            // Asset chosen but no pay value set state -------------------------------------------------------------------------------
+            root.swapFormData.fromTokenAmount = "1"
+            root.swapFormData.selectedAccountAddress = "0x7F47C2e18a4BBf5487E6fb082eC2D9Ab0E6d7240"
+            root.swapFormData.selectedNetworkChainId = 11155111
+            root.swapFormData.fromTokensKey = "ETH"
+            root.swapFormData.toTokenKey = ""
+
+            // Launch popup
+            launchAndVerfyModal()
+
+            const quoteItem = findChild(controlUnderTest, "quoteApproximationRight")
+            verify(!!quoteItem)
+
+            const sellItem = findChild(controlUnderTest, "quoteApproximationLeft")
+            verify(!!sellItem)
+
+            const priceItem = findChild(controlUnderTest, "quoteApproximationPrice")
+            verify(!!priceItem)
+
+            const invertQuoteApproximation = findChild(controlUnderTest, "invertQuoteApproximation")
+            verify(!!invertQuoteApproximation)
+
+            verify(!sellItem.visible)
+            verify(!quoteItem.visible)
+            verify(!priceItem.visible)
+            verify(!invertQuoteApproximation.visible)
+
+            root.swapFormData.toTokenKey = "STT"
+
+            fetchSuggestedRoutesCalled.wait()
```

Comment:
Hmm :thinking: You should really do without any `wait()` in tests...

Reply:
I'll have a look if we can ditch the wait(). But I'm afraid we'll just have to either wait for the signal to be emitted or use a `tryVerify`

Reply:
Yeah, `waitForItemPolished/Rendering()` + `tryCompare()` should do the job, and much better

Reply:
I've replaced it with tryCompare